lib/Infer/AliveDriver.cpp

```diff
@@ -890,20 +894,27 @@ struct RefinementProblem {
     Result.LHS = getInstCopy(LHS, IC, InstCache, BlockCache, &ConstMap, false);
     Result.RHS = getInstCopy(RHS, IC, InstCache, BlockCache, &ConstMap, false);
     Result.Pre = getInstCopy(Pre, IC, InstCache, BlockCache, &ConstMap, false);
+    Result.BPCs = BPCs;
+    for (auto &BPC : Result.BPCs) {
+      BPC.PC.LHS = getInstCopy(BPC.PC.LHS, IC, InstCache, BlockCache,
+                               &ConstMap, false);
+    }
 
     // Recursively call ReplacePhi, because Result might have Phi`s
     return Result.ReplacePhi(IC, Change);
   }
   bool operator == (const RefinementProblem &P) const {
-    return LHS == P.LHS && RHS == P.RHS && Pre == P.Pre;
+    return LHS == P.LHS && RHS == P.RHS &&
+           Pre == P.Pre && BPCs.size() == P.BPCs.size();
```

Comment:
This size() comparison is too loose and may cause tricky failures in future. Could we implement operator= for BPCs and checks the equality for each element?

Reply:
Good point. Being slightly slower to avoid tricky failures seems reasonable.